arch/xtensa/src/esp32/esp32_wifi_adapter.c

```diff
@@ -1165,6 +1188,70 @@ static int32_t esp_semphr_give(void *semphr)
   return osi_errno_trans(ret);
 }
 
+/****************************************************************************
+ * Name: esp_semphr_take_from_isr
+ *
+ * Description:
+ *   Try to take semaphore from within an interrupt service routine.
+ *
+ * Input Parameters:
+ *   semphr - Semaphore data pointer
+ *
+ * Returned Value:
+ *   True if success or false if fail
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_ESP32_WIFI_BT_COEXIST
```

Comment:
```suggestion
#ifdef CONFIG_ESP32_WIFI_BT_COEXIST

/****************************************************************************
 * Name: esp_semphr_take_from_isr
 *
 * Description:
 *   Try to take semaphore from within an interrupt service routine.
 *
 * Input Parameters:
 *   semphr - Semaphore data pointer
 *
 * Returned Value:
 *   True if success or false if fail
 *
 ****************************************************************************/

```
Since this `ifdef` wraps more than just a single function, it is better to move it up so that the documentation for the first function is not left out.